doc/chapters/debugging/index.rst

```diff
@@ -0,0 +1,197 @@
+
+.. _debugging:
+
+Debugging
+*********
+
+The use of containers can make debugging a bit harder as we sometimes need to
+get information from different sources of information. Paths can require some
+help to parse for the computer as well as yourself
+
+Debugging applications running in containers
+============================================
+
+This section and subsections describes some common ways of debugging
+applications running inside containers and how to use common tools on the
+applications running inside the containers.
+
+LXC
+---
+
+LXC provides several tools that can be used to check out the containers
+currently running on the system. `Note that all commands must be run as the main
+user which the containers are started as, ie root usually.`
+
+*lxc-ls* lists all currently running containers on the host.
+
+*lxc-attach* can be used to "get inside" the container with a shell for example,
+or to test run commands on the inside of the container. Please note that the
+command given must be available inside the container.
+
+*lxc-console* can be used to login with a shell inside the container in question.
+
+*lxc-info* provides information about the running container, such as network
+interface used for t he bridge, the state of the container, what PID it has and
+so forth.
+
+Cleanup
+-------
+
+The internal filesystem will always be deleted and recreated when an app is
+restarted. This means that anything you want to save during debugging needs to
+be saved to a bind mounted directory before you shut down the container.
+
+`SoftwareContainer` will also delete all container instances upon startup,
+meaning that container instances will be lost when restarting.
+
+GDB
+---
+
+Attaching a `gdb` session can be done in two ways, one is to run the `gdb` session
+and attach inside the container, the other is to attach to the process from the
+outside of the container. Using both methods have their own set of problems you
+need to solve.
+
+Running GDB inside container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you run `gdb` from inside the container using `lxc-attach`, you should not
+need to change any paths etc to find the correct libraries etc. You will need
+to have the `PID` from inside the container available to connect to however. this
+can be found out by running::
+
+    sudo lxc-attach --name SC-X ps aux
+
+Then running::
+
+    sudo lxc-attach --name SC-X -- gdb --pid YY
+
+Replace YY with the `PID` as seen inside the container. You can also run the
+application directly inside the container and start it with gdb directly::
+
+    sudo lxc-attach --name SC-X -- gdb --args /bin/ls
+
+TODO: Uncertain why, but this fails with unable to find executable.
```

Comment:
I don't think so. I had the same issue with other applications as well... and gdb does not read shell builtins in that way anyways afaik.